tsk2/tsk2/Car.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  Car.swift
+//  tsk2
+//
+//  Created by Ivan Markov on 03.03.2022.
+//
+
+import Foundation
+
+
+struct CarAttributes {
+    
+    var wheelsNum: Int
+    var doorsNum: Int
+
+    init(wheelsNum: Int, doorsNum: Int) {
+        self.wheelsNum = wheelsNum
+        self.doorsNum = doorsNum
+    }
+
+    init() {
+        wheelsNum = 0
+        doorsNum = 0
+    }
+}
+
+
+struct CarSize {
+    
+    let weight: Double
+    let width: Double
+    let length: Double
+    let height: Double
+    var volume: Double {
+        width * length * height
+    }
+}
+
+
+enum CarState {
+    
+    case settingUp
+    case working
+    case broken
+    case modified
+}
+
+
+class Car: NSCopying {
+
+    let modelName: String
+
+    let size: CarSize
+    
+    let defaultAttributes: CarAttributes
+    var actualAttributes: CarAttributes
+
+    let defaultEngine: String
+    var actualEngine: String
+
+    private let specialAction: () -> ()
+    
+    var state: CarState
+    
+    init(
+            modelName: String,
+            size: CarSize,
+            engine: String,
+            attributes: CarAttributes,
+            specialAction: @escaping () -> ()
+    ) {
+        self.modelName = modelName
+        self.size = size
+        defaultEngine = engine
+        actualEngine = ""
+        defaultAttributes = attributes
+        actualAttributes = CarAttributes()
+        self.specialAction = specialAction
+        state = .settingUp
+    }
+
+    init(
+            modelName: String,
+            size: CarSize,
+            defaultAttributes: CarAttributes,
+            actualAttributes: CarAttributes,
+            defaultEngine: String,
+            actualEngine: String,
+            specialAction: @escaping () -> (),
+            state: CarState
+    ) {
+        self.modelName = modelName
+        self.size = size
+        self.defaultAttributes = defaultAttributes
+        self.actualAttributes = actualAttributes
+        self.defaultEngine = defaultEngine
+        self.actualEngine = actualEngine
+        self.specialAction = specialAction
+        self.state = state
+    }
+
+    func assemble() {
+        actualAttributes = defaultAttributes
+        actualEngine = defaultEngine
+        state = .working
+    }
+
+    func crash() {
+        state = .broken
+    }
+
+    func crash(doorsAmount: Int) {
```

Comment:
В принципе посокльку все методы crash() ничего не возвращают а просто меняют данные самого класса, то можно эти методы сделать одним, просто добавив дефолтное значение параметрам
```suggestion
    func crash(doorsAmount: Int = 0, wheelsAmount: Int = 0) {
```

Reply:
Ну и так по аналогии все входящие параметры